### PR TITLE
Bitcoincore 0.12.1

### PR DESCRIPTION
**Pull request by luke-jr**
@blueness

### CODE REVIEW

**Comment by luke-jr** [comment]
Note that bitcoin-cli has no code changes between 0.12.0 and 0.12.1

File: eclass/bitcoincore.eclass [hunk 1]
```diff
@@ -198,7 +198,7 @@ fi
 if [ "${BITCOINCORE_NEED_LIBSECP256K1}" = "1" ]; then
 	BITCOINCORE_COMMON_DEPEND="${BITCOINCORE_COMMON_DEPEND} $LIBSECP256K1_DEPEND"
 fi
-if [ "${PN}" != "libbitcoinconsensus" ]; then
+if [ "${PN}" != "libbitcoinconsensus" ] && ! use_if_iuse test; then
```
Comment (blueness) [inline eclass/bitcoincore.eclass]: We're not reverting this.

File: eclass/bitcoincore.eclass [hunk 2]
```diff
@@ -198,7 +198,14 @@ fi
 if [ "${BITCOINCORE_NEED_LIBSECP256K1}" = "1" ]; then
 	BITCOINCORE_COMMON_DEPEND="${BITCOINCORE_COMMON_DEPEND} $LIBSECP256K1_DEPEND"
 fi
-if [ "${PN}" != "libbitcoinconsensus" ] && ! use_if_iuse test; then
+if [ "${PN}" = "libbitcoinconsensus" ]; then
```
Comment (blueness) [inline eclass/bitcoincore.eclass]: Looks like you reverted the use_if_iuse and then applied again.  Can you squash these two.

Comment (luke-jr) [inline eclass/bitcoincore.eclass]: Yes, it seemed cleaner this way. I can squash if desired.

Comment (blueness) [inline eclass/bitcoincore.eclass]: This would expose a commit where we have a QA problem, so yeah, squash them.

**Comment by blueness** [comment]
Okay I pushed this but there is one minor problem.  Because you add the test? ( ... ) dependency to BITCOINCORE_COMMON_DEPEND, it winds up in both DEPEND and RDEPEND.  The fact that you have a test? ( ... ) in RDEPEND causes repoman to throw a RDEPEND.suspect:

net-libs/libbitcoinconsensus/libbitcoinconsensus-0.12.1.ebuild: 'test?' USE conditional in RDEPEND

The dependency is required in DEPEND but not RDEPEND.  So its just a warning but you may want to clean that up.

**Comment by luke-jr** [comment]
It's actually a RDEPEND too, since libbitcoinconsensus[test] installs test_bitcoin.

**Comment by blueness** [comment]
okay so ignore the QA message.

**Pull request closed without merging**